sample.yml

```diff
@@ -30,12 +30,32 @@
     - dnsmasq
 
 # ROLES - after provisioning the software that's common to all hosts, we do
-# specialized hosts. This configuration has two major groups: control nodes and
-# worker nodes. We provision the worker nodes first so that we don't create any
-# race conditions. This could happen in the Mesos components - if there are no
-# worker nodes when trying to schedule control software, the deployment process
-# would hang.
-#
+# specialized hosts. There are 4 built-in roles: control, workers (mesos),
+# kubeworkers (kubernetes), and edge (for external traffic into the cluster).
+
+# The control nodes are necessarily more complex than the worker nodes, and have
+# ZooKeeper, Mesos, and Marathon leaders as well as Kubernetes master
+# components. In addition, they control Vault to manage secrets in the cluster.
+# These servers do not run applications by themselves, they only schedule work.
+# That said, there should be at least 3 of them (and always an odd number) so
+# that ZooKeeper can get and keep a quorum.
+- hosts: role=control
```

Comment:
note that this changes the order we provision the nodes. control nodes are configured before workers and kubeworkers. The old mesos race should be resolved by #1465 and changing the order eliminates a kubernetes race. In my testing (on gce), this has been working fine but keep an eye out for any issues dues to this change